src/tools/idlc/src/generator.c

```diff
@@ -395,12 +396,49 @@ idlc_generate(const idl_pstate_t *pstate, const idlc_generator_config_t *config)
   memset(&generator, 0, sizeof(generator));
   generator.path = file;
 
+  if(config->base_dir) {
+    char* rel_path = NULL;
+    char* old_dir = dir;
+    // If we run into a path error we can still recover and use existing dir
+    if(idl_relative_path(config->base_dir, pstate->sources->path->name, &rel_path) == IDLC_NO_MEMORY)
+      goto err_mkpath;
+    // If root is not a parent of file path, or there was an error resolving rel path
+    // TODO maybe find a common directory between the two and build from there?
```

Comment:
Regarding the TODO: I generally leaning towards being strict and predictable. Doing something smart using the common directory seems like it would be a likely cause of unexpected behaviours, and that's not nice when one is modifying the file system.